src/posts/introducing-inference-data/index.md

```diff
@@ -0,0 +1,98 @@
+
+At each of these steps the PPLs generate many arrays, and not only single dimension arrays, but multi dimensional arrays.
+The prior predictive distribution, the posterior distribution, the posterior predictive distribution, the loglikelihood, and even the input data
+are all numberical quantities that need to be captured and ordered to facilitate analysis.
+
+Often these arrays need to be passed to specialized functions in order to check for MCMC convergence, model comparison, or getting summaries from the posterior distribution.
+
+And once the individual statistician has completed their work, they may either want to save it to disk,
+or share it with a colleague.
+
+## The Modern PPL designer
+
+Now consider a PPL designer. PPL designers typically want to focus on the core of the Bayesian inference,
+Including working with graph representation and samplers.
+However for their PPL to be useful they must ensure their users must have easy access to diagnostics,
+visualizations, and model criticism tools.
+Suddenly the scope of their codebase grows, and duplicative functionality now exists across
+the open source Bayesian Ecosystem
+
+## Life before az.InferenceData
+
+Before `az.InferenceData` there wasn't a consistent object or method
+
+- Each PPL designer would have their own internal objects to store these
+  - The objects may not be serializable to disk
+- All adjacent workflow libraries would need
+- The APIs and interfaces for all of these were inconsistent making life challenging for end useres
+  - Folks using different PPLs would be in completely different ecosystems
+
+## How az.InferenceData works and its implementation
+
+`az.InferenceData` is an object that stores (nearly) of outputs of Bayesian Modeling in a consisntent manner.
+Under the hood `az.InferenceData` is largely a collection of Xarray objects, with some utility functions built in as well.
+Xarray was the natural choice, especially for MCMC inference, because its ability to store, index, and perform calculations
+over multidimensional outputs are routine task for computational Bayesian modelers.
+
+![InferenceData Architecture](https://python.arviz.org/en/stable/_images/InferenceDataStructure.png)
+
+## How this enables ArviZ
+
+With a consistent object that represents the outputs of PPLs, it became possible develop ArviZ,
+a unified library for exploratory analysis of Bayesian models.
+Now it didn't matter what PPL someone decided to use,
+as long as they follow the InferenceData specification ArviZ will know "what to do".
+This enables PPL designer to focus on the core of inference,
+and also ensures Bayesian Practitioners don't need to relearn the entire toolbox if they decide to switch PPLs.
+
+### The benefit of Xarray
+
+Xarray in turn simplifies the life of the ArviZ devs because we don't need to focus too much
+on the data store object, and instead we can focus on the statistical computations.
```

Comment:
In case it helps, 5 out of the 8 key features listed in https://xarray.dev/#features: the first 4 and the last are key to ArviZ design.

Reply:
Added